Live-Snap/Live-Snap/Controllers/FetchSnapcodeViewController.swift

```diff
@@ -90,18 +90,75 @@ class FetchSnapcodeViewController: UIViewController, UITextFieldDelegate {
+            bitmojiSnapcodeEndpointRequest.start { (snapcodeImage: UIImage?) in
+                
+                guard let snapcodeImage = snapcodeImage else { completion(nil); return }
+                    
+                let snapcodeImageView = UIImageView(frame: CGRect(x: 0, y: 0, width: 300, height: 300))
+                snapcodeImageView.image = snapcodeImage
+                
+                let bitmojiImageView = UIImageView(frame: CGRect(x: 0, y: 0, width: 190, height: 190))
+                bitmojiImageView.center = snapcodeImageView.center
+                bitmojiImageView.backgroundColor = UIColor.snapYellow
+                bitmojiImageView.layer.cornerRadius = bitmojiImageView.frame.width / 2.0
+                bitmojiImageView.clipsToBounds = true
+                bitmojiImageView.image = bitmojiImage
+                
+                completion(UIImage.imageFromUIViews(views: [snapcodeImageView, bitmojiImageView]))
+            }
+        }
+    }
+    
+    func showCannotGetSnapcodeError() {
+        print("Could not get snapcode")
+    }
+    
+    func showInvalidSnapchatUsernameError() {
+        print("Invalid snapchat username")
+    }
+    
     @objc func nextButtonWasPressed() {
         guard let username = usernameTextField.text else { return }
         
-        let endpointRequest = SnapcodeEndpointRequest(snapchatUsername: username)
-        endpointRequest.start { (snapcodeImage: UIImage?) in
-            if let snapcodeImage = snapcodeImage {
-                System.shared.snapcode = snapcodeImage
-                let selectPhotoViewController = SelectPhotoViewController()
-                System.shared.appDelegate().pageViewController?.setViewControllers([selectPhotoViewController], direction: .forward, animated: true, completion: nil)
-            } else {
-                // display failure case here
+        // Check if username is valid
+        IsValidSnapchatUsernameEndpointRequest(snapchatUsername: username).start { (isValidUsername: Bool) in
+            guard isValidUsername else {
+                DispatchQueue.main.async {
+                    self.showInvalidSnapchatUsernameError()
+                }
+                return
+            }
+            
+            // Get snapcode with bitmoji image in the middle
+            self.getSnapcodeWithBitmojiImage(username: username) { (bitmojiSnapcodeImage: UIImage?) in
+                
+                // Get regular snapcode with ghose in the middle
+                let snapcodeEndpointRequest = SnapcodeEndpointRequest(snapchatUsername: username)
+                snapcodeEndpointRequest.start { (snapcodeImage: UIImage?) in
+                    
+                    if let bitmojiSnapcodeImage = bitmojiSnapcodeImage {
+                        System.shared.snapcode = bitmojiSnapcodeImage
+                    } else if let snapcodeImage = snapcodeImage {
+                        System.shared.snapcode = snapcodeImage
```

Comment:
Note that a we can get a 800x800 snapcode image using a PNG link and only a 300x300 snapcode image using SVG link. If the bitmoji is not available, we might want to use the PNG image for better quality.

Reply:
if bitmoji is not available it won't be set here